core/domain/opportunity_domain.py

```diff
@@ -198,3 +198,97 @@ def validate(self):
                 'incomplete_translation, needs_voiceover and assigned_voiceover'
                 ' to be the same as the supported audio languages, '
                 'received %s' % list(expected_set_of_all_languages))
+
+
+class SkillOpportunity(python_utils.OBJECT):
+    """The domain object for skill opportunities."""
+
+    def __init__(
+            self, skill_id, skill_description,
+            question_count=0, topic_id=None, topic_name=None):
```

Comment:
I wasn't aware of that preference here, so I was going off the Python style guide linked in the docs [here](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#212-default-argument-values).

I figured that these would be reasonable default args since question_count initialized to 0 makes sense in most cases, and I was assuming that topic information would not always be available per skill.

Reply:
Ah, got it. Let's still not use default args where possible (this is more a code design preference than a code style preference). Better to force the caller to specify explicitly.

Also, note that SkillOpportunity objects can, in general, have varying numbers of questions, so there isn't really a reason to prefer 0 during initialization of this object.

Reply:
Done.